optimum/habana/diffusers/schedulers/scheduling_euler_discrete.py

```diff
@@ -224,19 +230,33 @@ def step(
                 "See `StableDiffusionPipeline` for a usage example."
             )
 
+        if self.hpu_opt and self.step_index is None:
+            self._init_step_index(timestep)
+
         # Upcast to avoid precision issues when computing prev_sample
         sample = sample.to(torch.float32)
 
-        sigma, sigma_next = self.get_params(timestep)
+        if self.hpu_opt:
```

Comment:
what's is this hpu_opt for? see **self.hpu_opt = False** in __init__, not found "self.hpu_opt = True" in any place.

Reply:
"self.hpu_opt = True" will be done in MLPERF SD XL repo where we instantiate the class. In optimum-habana, wont be visible. By default is set to Flase, so that current flow is intact in optimum-habana.

JH...!

Reply:
@ANSHUMAN87 do you mean optimum-habana doesn't need hpu_opt =true and no need for optimum-habana ? or you don't have method to set it to true for optimum-habana?

Reply:
As of now, optimum-habana no need to set to True, in case in future needed, user can instantiate the class and set to True, if user wants to use those optimizations. Those optimizations are applicable only for MLPERF SD-XL repo currently.